interop_binaries/tests/end_to_end.rs

```diff
@@ -0,0 +1,601 @@
+        port,
+    ))
+}
+
+/// RAII guard to ensure that child processes are cleaned up during test failures.
+struct ChildProcessCleanupDropGuard(Child);
+
+impl Drop for ChildProcessCleanupDropGuard {
+    fn drop(&mut self) {
+        if self.0.try_wait().unwrap().is_none() {
+            self.0.kill().unwrap();
+            self.0.wait().unwrap();
+        }
+    }
+}
+
+/// Pass output from a child process's stdout pipe to print!(), so that it can be captured and
+/// stored by the test harness.
+async fn forward_stdout(stdout: ChildStdout) -> io::Result<()> {
+    let mut reader = BufReader::new(stdout);
+    let mut line = String::new();
+    loop {
+        line.clear();
+        let count = reader.read_line(&mut line).await?;
+        if count == 0 {
+            return Ok(());
+        }
+        print!("{}", line);
+    }
+}
+
+/// Pass output from a child process's stderr pipe to eprint!(), so that it can be captured and
+/// stored by the test harness.
+async fn forward_stderr(stdout: ChildStderr) -> io::Result<()> {
+    let mut reader = BufReader::new(stdout);
+    let mut line = String::new();
+    loop {
+        line.clear();
+        let count = reader.read_line(&mut line).await?;
+        if count == 0 {
+            return Ok(());
+        }
+        eprint!("{}", line);
+    }
+}
+
+/// Take a VDAF description and a list of measurements, perform an entire aggregation using
+/// interoperation test binaries, and return the aggregate result. This follows the outline of
+/// section 4.7 of draft-dcook-ppm-dap-interop-test-design-00.
+async fn run(
+    vdaf_object: serde_json::Value,
+    measurements: &[serde_json::Value],
+    aggregation_parameter: &[u8],
+) -> anyhow::Result<serde_json::Value> {
+    // Start up a database testcontainer for each aggregator directly, and don't set up the schema.
+    let leader_db_container =
+        CONTAINER_CLIENT.run(RunnableImage::from(Postgres::default()).with_tag("14-alpine"));
+    let leader_postgres_port = leader_db_container.get_host_port_ipv4(5432);
+    let helper_db_container =
+        CONTAINER_CLIENT.run(RunnableImage::from(Postgres::default()).with_tag("14-alpine"));
+    let helper_postgres_port = helper_db_container.get_host_port_ipv4(5432);
+
+    // Pick four ports for HTTP servers.
+    let client_port = pick_unused_port().context("couldn't pick a port for the client")?;
+    let leader_port = pick_unused_port().context("couldn't pick a port for the leader")?;
+    let helper_port = pick_unused_port().context("couldn't pick a port for the helper")?;
+    let collector_port = pick_unused_port().context("couldn't pick a port for the collector")?;
+    assert_eq!(
+        BTreeSet::from([client_port, leader_port, helper_port, collector_port]).len(),
+        4,
+        "Ports selected for HTTP servers were not unique",
+    );
+
+    // Step 1: Create and start containers. (here, we just run the binaries instead)
```

Comment:
Editorial nit: numbered sequences in comments like this have a bad habit of getting out of sequence as new steps are added or steps are deleted. Might be best to just remove the numeration.